File: Languages/typescript/src/provider/MeanAnalysis.ts
import { SimpleTableAnalysis } from '../model/SimpleTableAnalysis'
import { UserInfo } from '../model/UserInfo';

// Classe de analise que implementa a interface "SimpleTableAnalysis"
export class MeanAnalysis implements SimpleTableAnalysis {
    
    // Método de interface, responsável por realizar obter a media dos valores de credit em uma lista de usuarios
    // @param userInfoList Lista de usuarios
    // @return Media dos valores de credit
    public analysis(userInfoList: Array<UserInfo>): number{

        var sum: number = 0.0;
        userInfoList.forEach(userInfo => sum += userInfo.getCredit() );

        return sum/(userInfoList.length);
    }
}

File: Languages/python3/src/provider/MeanAnalysis.py
from src.model.SimpleTableAnalysis import SimpleTableAnalysis
from src.model.UserInfo import UserInfo

# Classe de analise que implementa a interface "SimpleTableAnalysis"
class MeanAnalysis(SimpleTableAnalysis):

    # Método responsável por obter a media de valores de credit na lista de usuarios
    # @list Lista de usuarios
    # @return Média de valores da lista
    def analysis(self, userInfoList):
        vsum = 0

        for userInfo in userInfoList:
            vsum += userInfo.getCredit()

        return vsum/len(userInfoList)
